test/test_general.cpp

```diff
@@ -50,3 +50,13 @@ TEST(General, TestLineParsing) {
     EXPECT_EQ(data[1], "Entry2");
     EXPECT_EQ(data[2], "EscapedEntry1;EscapedEntry2");
 }
+
+TEST(General, TestUtilFunctions) {
+    std::vector<unsigned short> data{3, 4, 2, 1};
+    EXPECT_EQ(printPi(data), "(3,4,2,1)");
+
+    std::set<unsigned short> data_set1{3, 4, 2, 1};
+    std::set<unsigned short> data_set2{1, 2, 3};
+    EXPECT_EQ(idx(10, 4, 2, data_set1, data_set2), 120 + 6 + 4);
+    EXPECT_EQ(idx(10, 4, 2, data_set1, 4), 160 + 8 + 6);
```

Comment:
While I always appreciate additional tests, these tests
1) are not related to the PR
2) contain lots of magic numbers and it is not quite clear what is being tested here. Less magic numbers and/or some comments would help here